src/Package/Impl/Sql/Publish/SProcGenerator.cs

```diff
@@ -0,0 +1,186 @@
+            _coreShell = coreShell;
+            _pss = pss;
+            _fs = fs;
+        }
+
+        /// <summary>
+        /// Generates SQL scripts for the deployment of R code into SQL database.
+        /// </summary>
+        public void Generate(SqlSProcPublishSettings settings, EnvDTE.Project targetProject) {
+            var targetFolder = Path.Combine(Path.GetDirectoryName(targetProject.FullName), "R\\");
+            if (!_fs.DirectoryExists(targetFolder)) {
+                _fs.CreateDirectory(targetFolder);
+            }
+
+            var targetProjectItem = targetProject.ProjectItems.Item("R") ?? targetProject.ProjectItems.AddFolder("R");
+
+            if (settings.CodePlacement == RCodePlacement.Table) {
+                CreateRCodeTable(settings, targetProject, targetFolder, targetProjectItem);
+                CreatePostDeploymentScript(settings, targetProject, targetFolder, targetProjectItem);
+            }
+            CreateStoredProcedures(settings, targetProject, targetFolder, targetProjectItem);
+        }
+
+        /// <summary>
+        /// Create SQL file that defines table template that will hold R code
+        /// </summary>
+        private void CreateRCodeTable(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            var creatTableScriptFile = Path.Combine(targetFolder, CreateRCodeTableScriptName);
+            using (var sw = new StreamWriter(creatTableScriptFile)) {
+                sw.WriteLine(Invariant($"CREATE TABLE {settings.TableName}"));
+                sw.WriteLine("(");
+                sw.WriteLine(Invariant($"[{SProcColumnName}] NVARCHAR(64),"));
+                sw.WriteLine(Invariant($"[{RCodeColumnName}] NVARCHAR(max)"));
+                sw.WriteLine(")");
+            }
+            targetProjectItem.ProjectItems.AddFromFile(creatTableScriptFile);
+        }
+
+        /// <summary>
+        /// Generates SQL post deployment script that pushes R code into a table
+        /// as well as 
+        /// </summary>
+        private void CreatePostDeploymentScript(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            var postDeploymentScript = Path.Combine(targetFolder, PostDeploymentScriptName);
+
+            using (var sw = new StreamWriter(postDeploymentScript)) {
+                sw.WriteLine(Invariant($"INSERT INTO {settings.TableName}"));
+
+                for (int i = 0; i < settings.Files.Count; i++) {
+                    var filePath = settings.Files[i];
+
+                    var sprocName = settings.SProcNames[filePath];
+                    if (!string.IsNullOrEmpty(sprocName)) {
+                        var content = GetRFileContent(filePath);
+                        sw.Write(Invariant($"VALUES ('{sprocName}', '{content}')"));
```

Comment:
`sProcName` needs quote escaping as well. Probably worth making it a helper somewhere, actually.